common/fixation_entry_unique_index.go

```diff
@@ -0,0 +1,124 @@
+package common
+
+import (
+	"encoding/binary"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/store/prefix"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/lavanet/lava/common/types"
+)
+
+/*
+The fixationEntryUniqueIndex is a list which keeps all the latest version indices (e.g. indices without the version
+number suffix) of fixation entries of a particular module in the KVStore. To avoid a mix of indices between different
+kinds of fixation entries, the module that wishes to save a fixationEntryUniqueIndex must save it with a prefix that
+identifies it uniquely. For example, when the packages module want to add a new fixationEntryUniqueIndex to the store,
+it acesses the store using the key prefix: "packages_uniqueIndex_value".
+
+fixationEntryUniqueIndex structure:
+Entry {
+    uint64 id; 			    // a numerical index of the element (fixationEntryUniqueIndex[id] = unique_index)
+    string unique_index; 	// an entry unique index
+}
+*/
+
+// GetFixationEntryUniqueIndexCount get the total number of FixationEntryUniqueIndex
+func GetFixationEntryUniqueIndexCount(ctx sdk.Context, storeKey sdk.StoreKey, entryKeyPrefix string) uint64 {
+	store := prefix.NewStore(ctx.KVStore(storeKey), []byte{})
+	byteKey := types.KeyPrefix(entryKeyPrefix + types.UniqueIndexCountKey)
+	bz := store.Get(byteKey)
+
+	// Count doesn't exist: no element
+	if bz == nil {
+		return 0
+	}
+
+	// Parse bytes
+	return binary.BigEndian.Uint64(bz)
+}
+
+// SetFixationEntryUniqueIndexCount set the total number of FixationEntryUniqueIndex
+func SetFixationEntryUniqueIndexCount(ctx sdk.Context, storeKey sdk.StoreKey, entryKeyPrefix string, count uint64) {
+	store := prefix.NewStore(ctx.KVStore(storeKey), []byte{})
+	byteKey := types.KeyPrefix(entryKeyPrefix + types.UniqueIndexCountKey)
+	bz := make([]byte, 8)
+	binary.BigEndian.PutUint64(bz, count)
+	store.Set(byteKey, bz)
+}
+
+// AppendFixationEntryUniqueIndex appends a FixationEntryUniqueIndex in the store with a new id and update the count
+func AppendFixationEntryUniqueIndex(
+	ctx sdk.Context,
+	storeKey sdk.StoreKey,
+	cdc codec.BinaryCodec,
+	entryKeyPrefix string,
+	fixationEntryUniqueIndex types.UniqueIndex,
+) uint64 {
+	// Create the FixationEntryUniqueIndex
+	count := GetFixationEntryUniqueIndexCount(ctx, storeKey, entryKeyPrefix)
+
+	// Set the ID of the appended value
+	fixationEntryUniqueIndex.Id = count
+
+	store := prefix.NewStore(ctx.KVStore(storeKey), types.KeyPrefix(entryKeyPrefix+types.UniqueIndexKey))
+	appendedValue := cdc.MustMarshal(&fixationEntryUniqueIndex)
+	store.Set(GetFixationEntryUniqueIndexIDBytes(fixationEntryUniqueIndex.Id), appendedValue)
+
+	// Update FixationEntryUniqueIndex count
+	SetFixationEntryUniqueIndexCount(ctx, storeKey, entryKeyPrefix, count+1)
+
+	return count
```

Comment:
why return the previous size and not the latest size:  count+1?

Reply:
done